18F/identity-idp
# Try replacing NewRelic browser instrumentation with custom error handler

## 🛠 Summary of changes

Implements a frontend error logger, replacing the NewRelic browser instrumentation.

**Why?**

- Removes the large NewRelic loader snippet from every page
   - The snippet exceeds 50% of the markup size of the Sign In page (15kb of 27kb, before compression)
   - The snippet is out-of-date, in part because the latest version of the snippet is even larger in size
   - The snippet is rendered inline and is therefore not cacheable
   - The snippet is **render-blocking**
   - The practical impact here is a reduction of ~98.2%, from 6.84kb of uncacheable gzip-compressed JavaScript, to 124 bytes of cacheable brotli-compressed JavaScript.†~ **Edit (after revisions):** 99.0%, from 6.84kb to 67 bytes gzipped
- Consolidates logging
- Reduce number of third-party JavaScript scripts present in the page, to reduce vulnerability surface area
- Allows for greater control over how errors are logged, e.g. excluding errors occurring from browser extensions
- We don't use most of the frontend NewRelic features, and are largely interested in uncaught errors

## 📜 Testing Plan

1. Go to http://localhost:3000
2. `tail -n1 log/events.log`
3. See "Frontend error" log result with `name`, `message`, and `stack` properties
